pareto/strategic_water_management/strategic_produced_water_optimization.py

```diff
@@ -215,6 +218,489 @@ def get_strategic_model_unit_container():
+        _missing_sets = set(required_sets_with_option) - _df_sets_set
+        if len(_missing_parameters) > 0:
+            for param in _missing_parameters:
+                df_parameters[param] = {}
+            warning_message = "Warning: {} are given, but {} Parameter data is missing. The following parameters have been assumed empty:".format(
+                optional_tab_name, optional_tab_name
+            ) + str(
+                _missing_parameters
+            )
+            warnings.warn(warning_message)
+        if len(_missing_sets) > 0:
+            for s in _missing_sets:
+                df_sets[s] = {}
+            warning_message = "Warning: {} are given, but other {} Set data is missing. The following set has been assumed to be empty:".format(
+                optional_tab_name, optional_tab_name
+            ) + str(
+                _missing_sets
+            )
+            warnings.warn(warning_message)
+        if len(requires_at_least_one) > 0:
+            for requires_list in requires_at_least_one:
+                _included_params = set(requires_list) & _df_parameters_set
+                if len(_included_params) < 1:
+                    warning_message = "Warning: {} are given, but some piping and trucking arcs are missing. At least one of the following arcs are required, missing sets have been assumed to be empty:".format(
+                        optional_tab_name
+                    ) + str(
+                        requires_list
+                    )
+                    warnings.warn(warning_message)
+
+    # If there is specific parameter data for optional tab, the optional tab name is also required
+    if (
+        len(required_parameters_with_option & _df_parameters_set) > 0
+    ) and optional_tab_name not in df_sets:
+        raise MissingDataError(
+            'Essential data is incomplete. Parameters for {} are given, but the "{}" Set is not given. Please add and complete the following tab(s): {}'.format(
+                optional_tab_name, optional_tab_name, optional_tab_name
+            )
+        )
+    return (df_sets, df_parameters)
+
+
+# TODO: Move this function to a utility
+def check_config_dependent_data(
+    df_sets,
+    df_parameters,
+    config_argument,
+    config_required_sets,  # {config option 1: [set1, set2], config option 2: []}
+    config_required_params,
+):
+    # Error message
+    error_message = []
+
+    # Data error is raised if
+    _df_sets_set = set(df_sets)
+    _df_params_set = set(df_parameters)
+
+    required_sets = set(config_required_sets[config_argument])
+    _missing_sets = required_sets - _df_sets_set
+
+    if len(_missing_sets) > 0:
+        error_message.append(
+            "The config option {} has been selected. The following sets are required for this option and are missing:".format(
+                config_argument
+            )
+            + str(_missing_sets)
```

Comment:
```suggestion
            f"The config option {config_argument} has been selected. The following sets are required for this option and are missing:"
            + str(_missing_sets)
```
A minor thing, but I recommend using an [f-string](https://www.geeksforgeeks.org/formatted-string-literals-f-strings-python/) instead of the `.format()` syntax. f-strings are a lot more readable, in my opinion.